src/utils/breaches.ts

```diff
@@ -0,0 +1,173 @@
+    console.error(errMsg);
+    captureMessage(errMsg);
+    // @ts-ignore: function will be deprecated
+    return { verifiedEmails, unverifiedEmails };
+  }
+
+  const monitoredEmails = await getUserEmails(user.id);
+  verifiedEmails.push(
+    await bundleVerifiedEmails({
+      user,
+      email: user.primary_email,
+      recordId: user.id,
+      recordVerified: user.primary_verified,
+      allBreaches,
+    }),
+  );
+  for (const email of monitoredEmails) {
+    if (email.verified) {
+      verifiedEmails.push(
+        await bundleVerifiedEmails({
+          user,
+          email: user.primary_email,
+          recordId: email.id,
+          recordVerified: email.verified,
+          allBreaches,
+        }),
+      );
+    } else {
+      unverifiedEmails.push(email);
+    }
+  }
+
+  // get new breaches since last shown
+  for (const emailEntry of verifiedEmails) {
+    // /** @type {any[]} */
+    const newBreachesForEmail = emailEntry.breaches.filter(
+      (breach) => breach.AddedDate >= user.breaches_last_shown,
+    );
+
+    for (const newBreachForEmail of newBreachesForEmail) {
+      newBreachForEmail.NewBreach = true; // add "NewBreach" property to the new breach.
+      emailEntry.hasNewBreaches = newBreachesForEmail.length; // add the number of new breaches to the email
+    }
+  }
+
+  return { verifiedEmails, unverifiedEmails };
+}
+
+function addRecencyIndex(foundBreaches: HibpLikeDbBreach[]) {
+  // /**
+  //  * @type {any[]}
+  //  */
+  const annotatedBreaches: HibpLikeDbBreach[] = [];
+  // slice() the array to make a copy so before reversing so we don't
+  // reverse foundBreaches in-place
+  const oldestToNewestFoundBreaches = foundBreaches.slice().reverse();
+  oldestToNewestFoundBreaches.forEach((annotatingBreach, index) => {
+    const foundBreach = foundBreaches.find(
+      (foundBreach) => foundBreach.Name === annotatingBreach.Name,
+    );
+    annotatedBreaches.push(Object.assign({ recencyIndex: index }, foundBreach));
+  });
+  return annotatedBreaches.reverse();
+}
+
+type options = {
+  user: userType;
+  email: string;
+  recordId: number;
+  recordVerified: boolean;
+  allBreaches: HibpLikeDbBreach[];
+};
+async function bundleVerifiedEmails(
+  options: options,
+): Promise<BundledVerifiedEmails> {
+  const { user, email, recordId, recordVerified, allBreaches } = options;
+  const lowerCaseEmailSha = getSha1(email.toLowerCase());
+
+  // find all breaches relevant to the current email
+  const foundBreaches = await getBreachesForEmail(
+    lowerCaseEmailSha,
+    allBreaches,
+    true,
+    false,
+  );
+
+  // TODO: remove after migration MNTOR-978
+  // adding index to breaches based on recency
+  const foundBreachesWithRecency = addRecencyIndex(foundBreaches);
+
+  if (!user) {
+    const errMsg = "breachResolutionV2: subscriber cannot be undefined";
+    console.error(errMsg);
+    captureMessage(errMsg);
+
+    // @ts-ignore: function will be deprecated
```

Comment:
If you put `@deprecated` in a comment about the function definition then VSCode and other tooling will pick that up.

Reply:
Heads-up that that needs to be in a JSDoc comment, i.e. one that starts with `/**` rather than just `/*` or `//`.